marshal_test.go

```diff
@@ -0,0 +1,379 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package httprequest_test
+
+import (
+	"io/ioutil"
+
+	"github.com/juju/httprequest"
+
+	gc "gopkg.in/check.v1"
+	"gopkg.in/errgo.v1"
+)
+
+type marshalSuite struct{}
+
+var _ = gc.Suite(&marshalSuite{})
+
+var marshalTests = []struct {
+	about           string
+	urlString       string
+	method          string
+	val             func() interface{}
+	expectURLString string
+	expectBody      string
+	expectError     string
+}{{
+	about:     "struct with simple fields",
+	urlString: "http://localhost:8081/:F1",
+	val: func() interface{} {
+		t := struct {
+			F1 int    `httprequest:",path"`
+			F2 string `httprequest:",form"`
+		}{
+			F1: 99,
+			F2: "some text",
+		}
+		return &t
+	},
+	expectURLString: "http://localhost:8081/99?F2=some+text",
+}, {
+	about:     "struct with renamed fields",
+	urlString: "http://localhost:8081/:name",
+	val: func() interface{} {
+		t := struct {
+			F1 string `httprequest:"name,path"`
+			F2 int    `httprequest:"age,form"`
+		}{
+			F1: "some random user",
+			F2: 42,
+		}
+		return &t
+	},
+	expectURLString: "http://localhost:8081/some%20random%20user?age=42",
+}, {
+	about:     "fields without httprequest tags are ignored",
+	urlString: "http://localhost:8081/:name",
+	val: func() interface{} {
+		t := struct {
+			F1 string `httprequest:"name,path"`
+			F2 int    `httprequest:"age,form"`
+			F3 string
+		}{
+			F1: "some random user",
+			F2: 42,
+			F3: "some more random text",
+		}
+		return &t
+	},
+	expectURLString: "http://localhost:8081/some%20random%20user?age=42",
+}, {
+	about:     "pointer fields are correctly handled",
+	urlString: "http://localhost:8081/:name",
+	val: func() interface{} {
+		t := struct {
+			F1 *string `httprequest:"name,path"`
+			F2 *string `httprequest:"age,form"`
+			F3 *string `httprequest:"address,form"`
+		}{
+			F1: newString("some random user"),
+			F2: newString("42"),
+		}
+		return &t
+	},
+	expectURLString: "http://localhost:8081/some%20random%20user?age=42",
+}, {
+	about:     "MarshalText called on TextMarhsalers",
+	urlString: "http://localhost:8081/:param1/:param2",
+	val: func() interface{} {
+		v := struct {
+			F1 testMarshaler  `httprequest:"param1,path"`
+			F2 *testMarshaler `httprequest:"param2,path"`
+			F3 testMarshaler  `httprequest:"param3,form"`
+			F4 *testMarshaler `httprequest:"param4,form"`
+		}{
+			F1: "test1",
+			F2: (*testMarshaler)(newString("test2")),
+			F3: "test3",
+			F4: (*testMarshaler)(newString("test4")),
+		}
+		return &v
+	},
+	expectURLString: "http://localhost:8081/test_test1/test_test2?param3=test_test3&param4=test_test4",
+}, {
+	about:     "MarshalText called on TextMarhsalers",
```

Comment:
ditto. Also, is this test testing something different from the previous one?
If so, please express that in the about field, otherwise delete the test
as being redundant.